sites/geohub/src/lib/server/helpers/createDatasetLinks.ts

```diff
@@ -211,41 +213,81 @@ export const createDatasetLinks = async (
 		});
 
 		if (is_raster) {
-			const b64EncodedUrl = getBase64EncodedUrl(feature.properties.url);
-			feature.properties.links.push({
-				rel: 'cog',
-				type: 'application/json',
-				href: `${titilerUrl}`
-			});
-			feature.properties.links.push({
-				rel: 'info',
-				type: 'application/json',
-				href: `${titilerUrl}/info?url=${b64EncodedUrl}`
-			});
-			feature.properties.links.push({
-				rel: 'statistics',
-				type: 'application/json',
-				href: `${titilerUrl}/statistics?url=${b64EncodedUrl}`
-			});
-			feature.properties.links.push({
-				rel: 'tiles',
-				type: 'image/png',
-				href: `${titilerUrl}/tiles/WebMercatorQuad/{z}/{x}/{y}.png?url=${encodeURIComponent(
-					b64EncodedUrl
-				)}&scale=1&resampling=nearest&return_mask=true${algorithmId ? '' : '&bidx=1'}`
-			});
-			feature.properties.links.push({
-				rel: 'tilejson',
-				type: 'application/json',
-				href: `${titilerUrl}/WebMercatorQuad/tilejson.json?url=${encodeURIComponent(
-					b64EncodedUrl
-				)}&scale=1&resampling=nearest&return_mask=true${algorithmId ? '' : '&bidx=1'}`
-			});
-			feature.properties.links.push({
-				rel: 'algorithms',
-				type: 'application/json',
-				href: `${titilerUrl.replace('cog', 'algorithms')}`
-			});
+			console.log('225', 'is_raster');
+			// const b64EncodedUrl = getBase64EncodedUrl(feature.properties.url);
+			const b64EncodedUrl = feature.properties.url;
+
+			if (tool) {
+				feature.properties.links.push({
+					rel: 'cog',
+					type: 'application/json',
+					href: `${titilerUrl}`
+				});
+				feature.properties.links.push({
+					rel: 'info',
+					type: 'application/json',
+					href: `${titilerUrl}/info?url=${b64EncodedUrl}&algorithm=${tool}`
+				});
+				feature.properties.links.push({
+					rel: 'statistics',
+					type: 'application/json',
+					href: `${titilerUrl}/statistics?url=${b64EncodedUrl}&algorithm=${tool}`
+				});
+				feature.properties.links.push({
+					rel: 'tiles',
+					type: 'image/png',
+					href: `${titilerUrl}/tiles/WebMercatorQuad/{z}/{x}/{y}.png?url=${encodeURIComponent(
+						b64EncodedUrl
+					)}&scale=1&resampling=nearest&return_mask=true${algorithmId ? '' : '&bidx=1'}&algorithm=${tool}`
+				});
+				feature.properties.links.push({
+					rel: 'tilejson',
+					type: 'application/json',
+					href: `${titilerUrl}/WebMercatorQuad/tilejson.json?url=${encodeURIComponent(
+						b64EncodedUrl
+					)}&scale=1&resampling=nearest&return_mask=true${algorithmId ? '' : '&bidx=1'}&algorithm=${tool}`
+				});
+				feature.properties.links.push({
+					rel: 'algorithms',
+					type: 'application/json',
+					href: `${titilerUrl.replace('cog', 'algorithms')}`
+				});
```

Comment:
Can you remove these duplicated codes? It looks same with line 255-290 apart from info and statistics endpoint.

You can merge both of them. also `info` endpoint does not have algorithm query param.